sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py

```diff
@@ -0,0 +1,2322 @@
+
+        If a delete retention policy is enabled for the service, then this operation soft deletes the blob
+        and retains the blob for a specified number of days.
+        After the specified number of days, the blob's data is removed from the service during garbage collection.
+        Soft deleted blob is accessible through List Blobs API specifying include=deleted option.
+        Soft-deleted blob can be restored using Undelete API.
+
+        :param lease:
+            Required if the blob has an active lease. Value can be a LeaseClient object
+            or the lease ID as a string.
+        :type lease: ~azure.storage.blob.lease.LeaseClient or str
+        :param datetime if_modified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only
+            if the resource has been modified since the specified time.
+        :param datetime if_unmodified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only if
+            the resource has not been modified since the specified date/time.
+        :param str if_match:
+            An ETag value, or the wildcard character (*). Specify this header to perform
+            the operation only if the resource's ETag matches the value specified.
+        :param str if_none_match:
+            An ETag value, or the wildcard character (*). Specify this header
+            to perform the operation only if the resource's ETag does not match
+            the value specified. Specify the wildcard character (*) to perform
+            the operation only if the resource does not exist, and fail the
+            operation if it does exist.
+        :param int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
+        """
+        access_conditions = get_access_conditions(lease)
+        mod_conditions = get_modification_conditions(
+            if_modified_since, if_unmodified_since, if_match, if_none_match)
+        if not self.snapshot:
+            kwargs['delete_snapshots'] = "include"
+        try:
+            self._client.blob.delete(
+                timeout=timeout,
+                snapshot=self.snapshot,
+                lease_access_conditions=access_conditions,
+                modified_access_conditions=mod_conditions,
+                **kwargs)
+        except StorageErrorException as error:
+            process_storage_error(error)
+
+    def delete_blob_snapshots(
```

Comment:
I think probably we should merge delete_blob_snapshots into delete_blob, so that user could complete both in one I/O

Reply:
Done!